site/content/docs/custom-layout.md

```diff
@@ -0,0 +1,149 @@
+type: Blog
+created: 2023-03-02
+authors: [philippe-du-preez]
+---
+
+You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type. Then we will create a home page for listing all the pages of this type.
+
+Let's start with creating a new `Recipe` document type.
+
+## Create a custom Contentlayer document type
+
+In order to create our new `Recipe` document type, we need to go inside `.flowershow/contentlayer.config.ts` and add instructions to tell Contentlayer how to parse:
+
+- `name`: namespace
+- `filePathPattern`: input files (Files within this path do not need to specify their `Type` to `Recipe`)
+- `fields`: meta data fields
+- `computedFields`: derived meta data fields
+
+```javascript
+//contentlayer.config.ts
+// ...
+const recipeFields = {
+  created: { type: "date", required: true },
+  authors: {
+    type: "list",
+    of: { type: "string" },
+  },
+  layout: { type: "string", default: "recipe" },
+};
+
+const Recipe = defineDocumentType(() => ({
+  name: "Recipe",
+  filePathPattern: "recipes/!(index)*.md*",
+  contentType: "mdx",
+  fields: {
+    ...sharedFields,
+    ...recipeFields,
+  },
+  computedFields,
+}));
+
+export default makeSource({
+  // ...
+  documentTypes: [Recipe, Page],
+  // ...
+});
+```
+
+We can reuse the sharedFields (title, description etc.) as needed and add some custom fields, `recipeFields`, for our `Recipe` type. These fields will be accessible in the `RecipeLayout` we will be making. We do not have to include the `Type`, since we are adding the file in the `content/recipes` directory. We can then add the needed fields to the frontmatter of our recipe posts, like so:
+
+```bash
+---
+title: Homemade Lasagne Recipe
+description: Lasagne is a classic Italian dish that is perfect for any occasion. It is a hearty and delicious meal that can be enjoyed by everyone.
+created: 2022-11-29
+authors: [John Doe]
+---
+Lasagne is a classic Italian dish that is perfect for any occasion. It is a hearty and delicious meal ...
+```
+
+>See more on how to use the `Author` document type [here](https://flowershow.app/docs/blog#blog-authors)
+
+After creating our `Recipe` document type, we are ready to create a custom layout to display this.
+
+## Create a custom layout
+
+Now, we can create our `RecipeLayout.tsx` at `.flowershow/layouts` to receive and display the title, date and authors for the pages:
+
+```javascript
+export function RecipeLayout({ children, ...frontMatter }) {
+    const { created, title, authors } = frontMatter
```

Comment:
We could extract much more data here... Even all the ingredients so that they can be displayed always in the same place on the recipe page, e.g. some box at the top or on the left hand side. Take a look at some example sites with recipes. I'm not saying you should create something fancy, but just try to show users why creating custom layout may be useful at all 🙂